components/src/dynamo/thompson_router/router.py

```
@@ -0,0 +1,734 @@
+    reuse_budget: int = 0
+    tokens_in: int = 0
+    last_worker: int | None = None
+    ranking_score: float = 0.0
+    stickiness_score: float = 0.0
+    lints_features: Any = None  # np.ndarray stored at decision time for residual update
+
+
+class KvThompsonRouter:
+    """Two-term scoring router backed by native KvRouter (pyo3).
+
+    score(w) = λ₁ × ranking(w) + λ₂ × stickiness(w)
```

Ruff (0.15.10)

[warning] Docstring contains ambiguous `×` (MULTIPLICATION SIGN). Did you mean `x` (LATIN SMALL LETTER X)?

(RUF002)